Library/Homebrew/formula.rb

```diff
@@ -2673,6 +2673,8 @@ def inreplace(paths, before = nil, after = nil, audit_result = true, &block) # r
   def setup_home(home)
     # Don't let bazel write to tmp directories we don't control or clean.
     (home/".bazelrc").write "startup --output_user_root=#{home}/_bazel"
+    # Don't dirty the git tree for git clones.
+    (home/".gitignore").write "*"
```

Comment:
This impacts formulae tests that run `git add` as it now errors. Not sure if there is a way to restrict to build only. Otherwise will need to update formulae, e.g.
```
  ==> Testing git-series
  ==> git init
  Initialized empty Git repository in /private/tmp/git-series-test-20240423-56212-o7xmxo/.git/
  ==> git add test
  The following paths are ignored by one of your .gitignore files:
  test
  hint: Use -f if you really want to add them.
  hint: Turn this message off by running
  hint: "git config advice.addIgnoredFile false"
  ==> Testing git-series (again)
```

Reply:
@cho-m Yeh, making this build-only with a parameter probably makes sense 👍🏻

Reply:
> Bit odd `~/.gitignore` is being read here. Is `core.excludesFile` set or something?

It is possible this isn't working as intended then.
* In test, we use `HOME=testpath`, so `~/.gitignore` is often `<repo>/.gitignore`.
* In build, we use `HOME=buildpath/.brew_home`, so it may not actually be used. 

Sadly, this is reverse of what we want.

Reply:
Ohh right I had in my head for some reason that `.brew_home` existed in tests too. Makes sense being build-only then.

Reply:
One thing is whether we only care about `npm` or want to also deal with all `git` repos. `~/.gitignore` works for `npm`. `~/.config/git/ignore` works for `git`. However, neither works for both by default.

We would need to write two separate files to support both (in which case we may want to use `~/.npmignore` if only used by `npm`). Alternatively, we would need to write `core.excludesFile` into `~/.gitconfig` or `~/.config/git/config`, but the non-standard format (in-house ini variant) makes it less ideal to manually modify.

Reply:
Oops! Thanks for the quick revert @MikeMcQuaid, sorry didn't realize our `test` directory structure was different. Trying again with #17141